lib/Recur/RRuleIterator.php

```diff
@@ -127,55 +127,120 @@ public function isInfinite()
     }
 
     /**
-     * This method allows you to quickly go to the next occurrence after the
-     * specified date.
+     * This method allows you to quickly go to the next occurrence after the specified date.
      *
      * @param DateTimeInterface $dt
      */
     public function fastForward(DateTimeInterface $dt)
     {
+        // We don't do any jumps if we have a count limit as we have to keep track of the number of occurrences
         if (!isset($this->count)) {
-            do {
-                $diff = $this->currentDate->diff($dt);
-                switch ($this->frequency) {
-                    case 'hourly':
-                        $i = $diff->days * 24;
-                        break;
-                    case 'daily':
-                        $i = $diff->days;
-                        break;
-                    case 'weekly':
-                        $i = $diff->days / 7;
-                        break;
-                    case 'monthly':
-                        $i = $diff->days / 30;
-                        break;
-                    case 'yearly':
-                        $i = $diff->days / 365;
-                        break;
-                }
-                $i /= $this->interval;
-                $i /= 4;
-                $i = floor($i);
-                $i = max(1, $i);
-                do {
-                    $previousDate = clone $this->currentDate;
-                    $this->next($i);
-                } while ($this->valid() && $this->currentDate < $dt);
-
-                $this->currentDate = $previousDate;
-                // do one step to avoid deadlock
-                $this->next();
-            } while ($i > 5 && $this->valid() && $this->currentDate < $dt);
+            $this->jumpForward($dt);
         }
 
         while ($this->valid() && $this->currentDate < $dt) {
             $this->next();
         }
+    }
 
+    /**
+     * This method allows you to quickly go to the next occurrence before the specified date.
+     *
+     * @param DateTimeInterface $dt
+     */
+    public function fastForwardBefore(DateTimeInterface $dt)
+    {
+        // We don't do any jumps if we have a count limit as we have to keep track of the number of occurrences
         if (!isset($this->count)) {
-            // We don't know the counter at this point anymore
-            $this->counter = NAN;
+            $this->jumpForward($dt);
+        }
+
+        $previousDate = null;
+        while ($this->valid() && $this->currentDate < $dt) {
+            $previousDate = clone $this->currentDate;
+            $this->next();
+        }
+
+        isset($previousDate) && $this->currentDate = $previousDate;
+    }
+
+    /**
+     * Return the frequency in number of days.
+     *
+     * @return float|int|null
+     */
+    private function getFrequencyCoeff()
```

Comment:
What about moving this switch to an array? All of the value are constants, you could event define a new private const at the top of this file? 

The methods could then be simplified to: 

```
    private function getFrequencyCoeff()
    {
        return static::FREQUENCY_COEFFS[$this->frequency] ?? null;
    }
```

Reply:
Const array was introduced in PHP 5.6, we need to support >=5.5. To keep it simple, we can keep the switch